### PR TITLE
Luis add email notification to warning tracker

### PR DESCRIPTION
**Pull request by luisarevalo21**
# Description
This PR, first addressed an issue that crashed when opening the tracking modal. 
Next it added email integration, sending an email when a user receives more than 3 warnings. 
Different emails are sent when receiving a warning, and 'issuing a blue square' 

**Note**: a blue isn't assigned it only emails the user saying they received one. That'll be another PR. 

## Related PRS (if any):
This frontend PR is related to the #[1123](https://github.com/OneCommunityGlobal/HGNRest/pull/1123) backend PR .
…

## Main changes explained:
Created new helper functions to send an email. 
Fetched the team leads emails to cc them in the email when issuing a warning. 
Adjusted the filtering method to created a flag of when to send an email

## How to test:
**Before testing this branch, please follow the instructions from this PR, you must follow this exactly, and have a google auth key in order to send an email. **
https://github.com/OneCommunityGlobal/HGNRest/pull/829
If you have questions/issues with this step I'd be more than happy to help. 

**Another note:**
Ensure you have 2 emails one to receive the email and another to be cc'd.

1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Log as an Owner account. 
5. Wait for dashboard to load. 
6. Ensure the user you'll assign a warning is on a team, that has a manager on it. 
7. For testing purposes go to `warningsController.js` and change the `emailSender` to your email for both the issue warning and blue square. The first parameter will be the one who is being issued the warning. You'll add your email after a comma.  
Screenshot 2024-10-04 at 8 15 28 PM
Screenshot 2024-10-04 at 8 50 44 PM

8. Next add your second email to be cc'd in the `getUserRoleByEmail` Function. You can add your email in the recipients array. 
Screenshot 2024-10-04 at 8 17 11 PM
Screenshot 2024-10-04 at 8 50 52 PM

9. Next issue the warning, selecting a user who is on a team, and issue warning in the modal. 
10. Once you receive the email, issue a blue square to the same user. 
11. You'll receive two emails one for issuing a warning and another issuing a blue square. 
12. Everything else should, work, deleting, issuing another warning etc.

Issuing Warning: 

Screenshot 2024-09-28 at 8 52 29 PM

Issuing Blue square:
Screenshot 2024-09-28 at 8 54 17 PM

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/10e6a7ea-b4ca-428d-9d0d-51a48ea82fda

https://github.com/user-attachments/assets/6cdeeed7-e053-499b-9930-e70e7c6e8c4e

## Note:
Before starting, to review this PR, you must have a google auth key to send emails. Follow the PR attached, and feel free to reach out if you get stuck.

### CODE REVIEW

File: src/actions/warnings.js [hunk 1]
```diff
@@ -35,10 +40,10 @@ export const postWarningByUserId = warningData => {
       const response = await dispatch(postWarningsByUserId(res.data));
       return response.payload.warnings;
     } catch (error) {
-      if (error.response && error.response.status === 400) {
-        return { error: 'Error occured when posting' };
+      if (error.response && error.response.status === 200) {
```
Comment (Ankuriboh) [inline src/actions/warnings.js:43]: Errors should never get a `200`, this also requires backend changes.

File: src/components/Warnings/modals/WarningTrackerModal.jsx [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+      setWarningWasEdited(true);
+    });
+  };
+
+  // eslint-disable-next-line no-shadow
+  const handleAddNewWarning = (e, newWarning) => {
+    e.preventDefault();
+
+    if (newWarning === '') return;
+    const trimmedWarning = newWarning.trim();
```
Comment (Ankuriboh) [inline src/components/Warnings/modals/WarningTrackerModal.jsx:183]: I would suggest adding frontend validation for special characters, even though backend already had it.

**Review by Ankuriboh — CHANGES_REQUESTED** [review]
Can confirm that emails were sent, and the modal after clicking circles in the tracking dropdown on the dashboard does not crash the app.

Some trivial things can be improved, see my code comments and also the corresponding backend PR.

**Comment by anji-annavarapu** [comment]
Screenshot 2024-10-23 135616


Hey, I am trying to follow the steps in a YouTube video for OAuth authentication. I got to the 'app isn't verified' step, but in the video, there is an option to continue by clicking 'Advanced.' However, I don't see this option in my case. I have attached a screenshot for your reference. Please help if you have any ideas.

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**